Java/avg/array.java

```diff
@@ -12,4 +12,15 @@ public static void main(String[] args) {
             System.out.println(arr[i]);
         }
     }
+
+    public static void Even(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        int[] arr = new int[]{1, 2, 3, 4, 5};
+        System.out.println("Even numbers:");
+
+        for (int i = 0; i < arr.length; i = i * 2)
+        {
+            System.out.println(arr[i]);
+        }
+    }

```

Comment:
**Code Structure Issue**: The Even method duplicates functionality with minor differences. Consider refactoring to reduce code duplication and enhance maintainability. <br> **Fix**: Refactor the common logic into a separate method that accepts a parameter to differentiate between odd and even number processing. <br> **Code Suggestion**: 
 ```
 public static void printNumbers(boolean even) {
     int[] arr = new int[]{1, 2, 3, 4, 5};
     System.out.println(even ? "Even numbers:" : "Numbers:");
     for (int i = even ? 0 : 1; i < arr.length; i += 2) {
         System.out.println(arr[i]);
     }
 }
 ```